src/EscapeAnalysis.jl

```diff
@@ -250,10 +251,14 @@ function find_escapes(ir::IRCode, nargs::Int)
                     if isa(lhs, GlobalRef)
                         add_change!(rhs, ir, Escape(), changes)
                     end
+                elseif is_meta_expr_head(head)
+                    continue
                 elseif head === :enter || head === :leave || head === :pop_exception
                     continue
+                elseif head === :gc_preserve_begin || head === :gc_preserve_end
+                    continue
                 else # TODO: this is too conservative
-                    add_changes!(stmt.args[2:end], ir, Escape(), changes)
+                    add_changes!(stmt.args, ir, Escape(), changes)
```

Comment:
I wonder why this changes? under the current master it will report an error on `ERROR: AssertionError: argextype only works on argument-position values`.

Reply:
We don't handle the expression yet, so escape all the arguments conservatively.

Reply:
We need to fix the assertion error separately.

Reply:
By adding an argument-position check before the argextpye call?

Reply:
Not sure honestly. Please debug where the issue stems from.

Reply:
It seems like some `:foreigncall` expression can contain nested `:call` expression, which causes this issue: e.g. `(16, :($(Expr(:foreigncall, :(Core.tuple(:pcre2_get_error_message_8, Base.PCRE.PCRE_LIB)), Int32, svec(Int32, Ptr{UInt8}, UInt64), 0, :(:ccall), Core.Argument(2), :(%15), :(%10), :(%10), :(%1), Core.Argument(2)))))`

Reply:
We find the same(?) similar place! Does this foreign call get turned into a call? You can see my log below

Reply:
OK, let me try to support `foreigncall`, looks not that hard.